### ISSUE TITLE
`Style/ComparableBetween` auto-correction can be unsafe

### ISSUE THREAD

**Issue by jasonkarns** [rubocop/rubocop#14082]
The `Style/ComparableBetween` cop is marked as safe for autocorrection but can be unsafe. (It presumes objects that respond to `<` or `>` also implement `#between?`.)

--------

## Expected behavior

I expected rubocop to not not break code when autocorrecting safe-cops. (sample ruby file in steps-to-reproduce)

## Actual behavior

Rubocop doesn't detect when `between?` is not implemented. So it's possible to have objects that respond to less-than, greater-than and attempt to autocorrect to use `between?`, but there is nothing to ensure those objects respond to `between?`.

## Steps to reproduce the problem

1. Given sample ruby script that demonstrates a ruby class that responds to `<=` and `>=`, but does not implement `Comparable` (thus it doesn't not respond to `between?`).
   ```ruby
   # comparable_between.rb
   # frozen_string_literal: true

   # Foo implements lt/gt operators but not Comparable
   class Foo
     def <=(*) = true
     def >=(*) = true
   end

   x = Foo.new
   p(x >= 1 && x <= 5)
   ```

2. The script is valid and runs successfully (prints true).
   ```
   $ ruby comparable_between.rb 
   true
   ```
3. After rubocop autocorrects
   ```
   $ rubocop -a --enable-pending-cops comparable_between.rb 
   Inspecting 1 file
   C

   Offenses:

   comparable_between.rb:10:3: C: [Corrected] Style/ComparableBetween: Prefer x.between?(1, 5) 
   over logical comparison.
   p(x >= 1 && x <= 5)
     ^^^^^^^^^^^^^^^^

   1 file inspected, 1 offense detected, 1 offense corrected
   ```
4. Then the script is no longer runs successfully
   ```
   $ ruby comparable_between.rb 
   comparable_between.rb:10:in `<main>': undefined method `between?' for # <Foo:0x0000000102edb270> (NoMethodError)

   p(x.between?(1, 5))
      ^^^^^^^^^
   ```

## RuboCop version

```
$ rubocop -V
1.75.2 (using Parser 3.3.7.4, rubocop-ast 1.44.0, analyzing as Ruby 3.2, running on ruby 3.2.6) [arm64-darwin24]
```
